### PR TITLE
feat(v3-parser): Export a return item for method items (#37)

### PR DESCRIPTION
**Pull request by soft-decay**

### CODE REVIEW

File: examples/Button.json [hunk 1]
```diff
@@ -127,7 +127,39 @@
...
+            "visibility": "public",
+            "description": "Computes the answer to your question.",
+            "name": "computeAnswer",
+            "args": [
+                {
+                    "name": "question"
+                }
+            ],
+            "return": {
+                "doctype": {
```
Comment (alexprey) [inline examples/Button.json:151]: This field must be called a `type` instead of `doctype`, like it used everywhere

File: examples/Button.json [hunk 2]
```diff
@@ -127,7 +127,39 @@
...
+                    "kind": "type",
+                    "text": "number",
+                    "type": "number"
+                },
+                "type": "number",
+                "desc": "the answer to all your questions",
```
Comment (alexprey) [inline examples/Button.json:157]: Not sure that we need to duplicate desciption for return. So, please keep only second property `description`

File: lib/jsdoc.js [hunk 3]
```diff
@@ -167,14 +168,41 @@ function parseParamKeyword(text) {
 }
 
 function parseReturnKeyword(text) {
-    const output = { type: DEFAULT_TYPE, desc: '' };
+    const output = {
+        doctype: {
```
Comment (alexprey) [inline lib/jsdoc.js:172]: Ok, I see that we have some old code with non consistent data structure. However, I'm not sure that it is a good idea to introduce the new structure of output data. 
I prefer to keep clean and consistent API and release this fix with new major version. What do you think about that?

Let's keep output in the following format: 
```
{
    type: { ... },
    description: '...'
}
```

File: lib/v3/parser.js [hunk 4]
```diff
@@ -385,8 +386,7 @@ class Parser extends EventEmitter {
                     }
 
                     if (declaration.type === 'FunctionDeclaration') {
-                        const func = this.parseFunctionDeclaration(declaration);
-
+                        const func = this.parseFunctionDeclaration(declaration, exportNodeComment.keywords);
```
Comment (alexprey) [inline lib/v3/parser.js:389]: `parseFunctionDeclaration` method used there only for parse JS AST nodes and provide low-level information that can be extracted only from JS AST. And then I use `emitMethodItem` method that use contextual information and fill additional data to item. So, I prefer to move `return` parsing logic to `emitMethodItem` method, that should cover all usages of the code and you is not need to extend method, becase second method is already has information about comment with all parsed keywords.

**Review by alexprey — CHANGES_REQUESTED** [review]
Hi! Thanks for PR, you found a very serious issue and solve it, I'm like that.
Please, make code changes to keep code clear and I can merge that

File: lib/jsdoc.js (hunk 3 above)
Comment (soft-decay) [reply lib/jsdoc.js:172]: Yes definitely, a major release would be the best option. I did not want to break anything so I introduced new keys in the current format. But if you're willing to do a major release, the old format of `method.return`:

```typescript
interface SvelteMethodReturnItem {
    type: string|string[];
    desc: string;
}
```
would become this:

```typescript
interface SvelteMethodReturnItem {
    type: JSDocType;
    description: string;
}

File: lib/v3/parser.js (hunk 4 above)
Comment (soft-decay) [reply lib/v3/parser.js:389]: Yes of course, it is a better separation of concerns.

**Review by soft-decay — COMMENTED** [review]
I commited the changes suggested.

**Review by alexprey — APPROVED** [review]
Looks great!

**Pull request merged**